File: gfg/march2024/24_insert_at_bottom_of_stack.py
# CODE

class Solution:
    def insertAtBottom(self, st, x):
        
        # Cheating
        # st.insert(0, x)
        # return st
        
        def solve(st, x):
            if len(st) == 0:
                st.append(x)
                return 
            val = st[-1]
            st.pop()
            solve(st, x)
            st.append(val)
        solve(st, x)
        return st

File: gfg/march2024/c++/24_insert_at_bottom_of_stack.cpp
// CODE

class Solution{
    public:
    void solve(stack<int> &st, int x) {
        if(st.empty()) {
            st.push(x);
            return ;
        }
        int val = st.top();
        st.pop();
        solve(st, x);
        st.push(val);
    }
    stack<int> insertAtBottom(stack<int> st, int x) {
        solve(st, x);
        return st;
    }
};